sys/decl.go

```diff
@@ -236,10 +236,21 @@ func (t FilenameType) Align() uintptr {
 	return 1
 }
 
+type ArrayKind int
+
+const (
+	ArrayRandLen ArrayKind = iota
+	ArrayFixedLen
```

Comment:
Yes.
All code out there should handle range arrays anyway, and the special case of degenerate range (fixed-size array).